src/rustdoc.md

```diff
@@ -36,30 +36,174 @@ does is call the `main()` that's in this crate's `lib.rs`, though.)
+Certain browser features that require secure origins, like `localStorage` and
+Service Workers, don't work reliably. We can still use such features but we
+should make sure pages are still usable without them.
+
+## Multiple runs, same output directory
+
+Rustdoc can be run multiple times for varying inputs, with its output set to the
+same directory. That's how cargo produces documentation for dependencies of the
+current crate. It can also be done manually if a user wants a big
+documentation bundle with all of the docs they care about.
+
+HTML is generated independently for each crate, but there is some cross-crate
+information that we update as we add crates to the output directory:
+
+ - crates.js holds a list of all crates in the output directory.
+ - search-index.js holds a list of all searchable items.
+ - For each trait, there is a file under `implementors/.../trait.TraitName.js`
+   containing a list of implementors of that trait. The implementors may be in
+   different crates than the trait, and the JS file is updated as we discover
+   new ones.
+
+## Use cases
+
+There are a few major use cases for rustdoc that you should keep in mind when
+working on it:
+
+### Standard library docs
+
+These are published at https://doc.rust-lang.org/std as part of the Rust release
+process. Stable releases are also uploaded to specific versioned URLs like
+https://doc.rust-lang.org/1.57.0/std/. Beta and nightly docs are published to
+https://doc.rust-lang.org/beta/std/ and https://doc.rust-lang.org/nightly/std/.
+The docs are uploaded with the [promote-release
+tool](https://github.com/rust-lang/promote-release) and served from S3 with
+CloudFront.
+
+The standard library docs contain five crates: alloc, core, proc_macro, std, and
+test.
+
+### docs.rs
+
+When crates are published to crates.io, docs.rs automatically builds
+and publishes their documentation, for instance at
+https://docs.rs/serde/latest/serde/. It always builds with the current nightly
+rustdoc, so any changes you land in rustdoc are "insta-stable" in that they will
+have an immediate public effect on docs.rs. Old documentation is not rebuilt, so
+you will see some variation in UI when browsing old releases in docs.rs. Crate
+authors can request rebuilds, which will be run with the latest rustdoc.
+
+Docs.rs performs some transformations on rustdoc's output in order to save
+storage and display a navigation bar at the top:
+
+ - The `<body>` tag is renamed to a `<div>` with the id `rustdoc_body_wrapper`
+   so docs.rs can have its own body tag.
+ - Static files (JS and CSS) are renamed so they unique identifiers (including a
+   hash) for each version, and stored independently of the crate docs.
+
+Documentation on docs.rs is always generated for a single crate at a time, so
+the search and sidebar functionality don't include dependencies of the current
+crate.
+
+### Locally generated docs
+
+Develops run `cargo doc --open` in crates they have checked out locally to see
```

Comment:
"Developers" seems to mean "developers of rustdoc", but that's not what you mean AFAICT:

```suggestion
Users can run `cargo doc --open` in crates they have checked out locally to see
```